src/extension.ts

```diff
@@ -1,98 +1,175 @@
-import * as vscode from 'vscode';
-
-import { TelemetryEvent } from './TelemetryEvent';
-import { TelemetryEventQueue } from './TelemetryEventQueue';
-import { Reporter } from './Reporter';
-import { Logger } from './Logger';
-
-const OPTIN_REQUESTED_KEY = 'redhat.telemetry.optInRequested';
-
+import * as vscode from "vscode";
+import { SegmentInitializer } from "./segmentInitializer";
+import { Logger } from "./utils/logger";
+import Analytics from "analytics-node";
+import { TelemetryEventQueue } from "./utils/telemetryEventQueue";
+import { TelemetryEvent } from "./interfaces/telemetryEvent";
+import { Reporter } from "./reporter";
+
+/* 
+  OPT_IN_STATUS == "true" if user Agreed
+  false when denied for telemetry data collection     
+*/
+const OPT_IN_STATUS = "redhat.telemetry.optInRequested";
+
+// this method is called when your extension is activated
 export function activate(context: vscode.ExtensionContext) {
-  Reporter.initialize(context);
+  Logger.log('"vscode-commons" is now active!');
+
+  //  instance of initialized segment object
+  const analytics: Analytics | undefined = SegmentInitializer.initialize(
```

Comment:
ultimately the Analytics instance should be dependent on the segmentKey from the caller extension's context. If that extension has no segmentKey, then fall back to using vscode-commons'

Reply:
Okay, Updated the PR with his change.